### PR TITLE
Change trtl window, fix window bug

### PR DESCRIPTION
**Pull request by almogdepaz**

### CODE REVIEW

File: tortoise/ninja_tortoise.go [hunk 1]
```diff
@@ -502,7 +503,7 @@ func (ni *ninjaTortoise) handleIncomingLayer(newlyr *types.Layer) { //i most rec
 			if Window > newlyr.Index() {
 				windowStart = 0
 			} else {
-				windowStart = newlyr.Index() - Window + 1
+				windowStart = Max(ni.pBase.Layer()+1, newlyr.Index()-Window+1)
```
Comment (zalmen) [inline tortoise/ninja_tortoise.go:506]: Is it possible that Pbase will be bellow the window?

File: tortoise/ninja_tortoise.go [hunk 2]
```diff
@@ -533,7 +534,7 @@ func (ni *ninjaTortoise) handleIncomingLayer(newlyr *types.Layer) { //i most rec
 			}
 
 			complete := true
-			for idx := windowStart; idx < j; idx++ {
+			for idx := types.LayerID(0); idx < j; idx++ {
```
Comment (zalmen) [inline tortoise/ninja_tortoise.go:537]: why start from layer 0? We shouldn't hold any info on layers outside the window, so why not start from the bottom of the window? Perhaps I'm missing the purpose of this iteration.

Comment (almogdepaz) [reply tortoise/ninja_tortoise.go:537]: true this was a temporary solution, we should only save the decision on blocks outside of window, we need to clean the maps and save a boolean regarding contextual validity

Comment (almogdepaz) [reply tortoise/ninja_tortoise.go:537]: of course this part has a lot to do with the layer roleback and maybe we should think  about it considering the changes we still have to make let me know how you want to proceed with this

File: tortoise/ninja_tortoise.go (hunk 1 above)
Comment (almogdepaz) [reply tortoise/ninja_tortoise.go:506]: it is but it shouldn't happen that means that pbase is stuck and we need to do a roleback/selfhealing/somthing

File: tortoise/ninja_tortoise.go (hunk 2 above)
Comment (almogdepaz) [reply tortoise/ninja_tortoise.go:537]: #803

File: tortoise/ninja_tortoise.go [hunk 3]
```diff
@@ -558,7 +559,10 @@ func (ni *ninjaTortoise) handleIncomingLayer(newlyr *types.Layer) { //i most rec
 						complete = false //not complete
 					}
 				}
-				updatePatSupport(ni, p, bids, idx)
+
```
Comment (antonlerner) [inline tortoise/ninja_tortoise.go:562]: add ut for this case

**Review by barakshani — APPROVED** [review]

**Pull request merged**